bin/node/Cargo.toml

```diff
@@ -19,7 +19,7 @@ tracing-forest = ["miden-node-block-producer/tracing-forest"]
 
 [dependencies]
 anyhow = { version = "1.0" }
-clap = { version = "4.5", features = ["derive"] }
+clap = { version = "4.5", features = ["derive", "string"] }
```

Comment:
Question: why do we need to add `string` dependency here, but we don't need to do the same for the faucet binary?

Reply:
It is needed by both but its included automatically due to feature unification. Removing it causes both `node` and `faucet` to fail, including it in either then both work.

Adding to `faucet` as well since that's more correct.

Reply:
To be more specific, building with individual feature resolution would fail.

```
# This works as by default features are unified at workspace level.
cargo build --bin miden-faucet

# This fails
cargo build -p miden-faucet --bin miden-faucet
```

Maybe there is something to be said about building each binary in isolation in CI?

Reply:
Hmmm - don't have a strong opinion here.

Reply:
Looks like this is somewhat easy to perform using [cargo hack](https://github.com/taiki-e/cargo-hack?tab=readme-ov-file#--all---workspace).

If one just uses `check` then it shouldn't take much time to perform for each package. Not terribly important though.

Reply:
#501